### ISSUE TITLE
Green~Red (looping) in group mode Release issue

### ISSUE THREAD

**Issue by Airell** [qiemem/eurorack#51]
Documented is that Green+Red in a group acts like a Turing Machine, so when in a 'ADSR' group (Green/Green/Green+Red/Green) it will 'choose' a Sustain level (? as it seems). However the Release part (last Green segment) does not start to lower at the choosen sustain level. It might jump up to a higher level first, before it releases.

Also in a group, the Green+Red (not looping) needs a new trigger to move to the Release segment. When using Green+Red (looping) in a segment, it will move to the Release segment on the release if the Attack trigger.

**Comment by qiemem** [comment]
> Documented is that Green+Red in a group acts like a Turing Machine, so when in a 'ADSR' group (Green/Green/Green+Red/Green) it will 'choose' a Sustain level (? as it seems). However the Release part (last Green segment) does not start to lower at the choosen sustain level. It might jump up to a higher level first, before it releases.

Thanks for the report! I believe that was fixed in this commit: d45e366c13ccd59f7a6b89f85da8f925449a1089... which alas hasn't been included in an official release. It *should* be in any of the builds on this thread: https://github.com/qiemem/eurorack/issues/48#issuecomment-1426874557

> Also in a group, the Green+Red (not looping) needs a new trigger to move to the Release segment. When using Green+Red (looping) in a segment, it will move to the Release segment on the release if the Attack trigger.

Yes, that's intentional. This way you can have a sequence of shift registers that you can step through, or use it for some random sustain values. To be honest, I've never been super satisfied with the behavior of random segments in groups. I was trying to kind of get the best of both worlds, but it sort of just feels inconsistent. So if there's behavior you're looking for here, I'm to suggestions!

**Comment by Airell** [comment]
Superb, thanks!
As you probably noticed from an other comment on #48, I did encounter this on the v1.1.0 version, as I thought this was the latest labeled release. I was trying to figure it out, with the shift register in a group of Greens surrounding, when I noticed it. Later I found the (very satisfying) PDF which the shift register step and hold in a group on the last page.

I do can think of a fully shift register group where you only have shift register's. It will act more like a 'turing machine', where the previous segment value is moved to the next segment on a gate signal in the first segment. I.e. "G/R gate, G/R looping, G/R looping, G/R looping". Maybe it already works this way, I haven't tried 😅
